### PR TITLE
Don't try importing empty - but formatted - rows

### PR DESCRIPTION
**Pull request by esoergel**
https://manage.dimagi.com/default.asp?283896

The project in the above FB ticket apparently had a spreadsheet which openpyxl showed as having 1,048,553 rows, though it clearly had only two.  That's just shy of the max number of rows allowed in xlsx files.  My best guess is that there is some sort of formatting that's applied globally, so _every_ possible row counted as a row.  Openpyxl considers this a faithful reproduction of the xlsx spec, and leaves it to projects to work around ([issue](https://bitbucket.org/openpyxl/openpyxl/issues/514/cell-max_row-reports-higher-than-actual)).  To support that hypothesis, I copied the first two rows from that project's spreadsheet into a new spreadsheet and got the expected number of max rows.

This change cuts the file short after the last row with data in it.  Notably, I didn't do anything for empty rows in the middle of a file, since that felt like it could have unintended consequences.

### CODE REVIEW

File: corehq/util/workbook_reading/adapters/xlsx.py [hunk 1]
```diff
@@ -59,13 +60,27 @@ def _make_cell_value(self, cell):
...
+
+    def iter_rows(self):
+        for i, row in enumerate(self._worksheet.iter_rows(), 1):
+            if i > self._max_row:
+                break
             yield [Cell(self._make_cell_value(cell)) for cell in row]
 
     def to_worksheet(self):
         # Note that an empty sheet and a sheet with one row both have max_row = 1
-        return Worksheet(title=self._worksheet.title, max_row=self._worksheet.max_row,
```
Comment (nickpell) [inline corehq/util/workbook_reading/adapters/xlsx.py]: Should this be updated as well?

https://github.com/dimagi/commcare-hq/blob/57f56f1c5ab1243126057ad5eeb3c5fe7a1868e9/corehq/apps/case_importer/util.py#L127-L129

File: corehq/util/workbook_reading/adapters/xlsx.py [hunk 2]
```diff
@@ -59,13 +60,27 @@ def _make_cell_value(self, cell):
...
+        # Note that this is 1-indexed for consistency with openpyxl
+        max_row = 1
         for row in self._worksheet.iter_rows():
+            if any(cell.value for cell in row):
+                max_row = row[0].row
+        return max_row
+
+    def iter_rows(self):
+        for i, row in enumerate(self._worksheet.iter_rows(), 1):
+            if i > self._max_row:
```
Comment (nickpell) [inline corehq/util/workbook_reading/adapters/xlsx.py]: If ```self._max_row``` is 1-indexed, should this be ```>=```?

**Review by nickpell — COMMENTED** [review]
Nice find!

File: corehq/util/workbook_reading/adapters/xlsx.py (hunk 2 above)
Comment (esoergel) [reply corehq/util/workbook_reading/adapters/xlsx.py]: I did `enumerate(..., 1)` so they'd be consistent

File: corehq/util/workbook_reading/adapters/xlsx.py (hunk 1 above)
Comment (esoergel) [reply corehq/util/workbook_reading/adapters/xlsx.py]: I don't think so, I believe `self._worksheet` there is created here:

https://github.com/dimagi/commcare-hq/blob/c6155325492cdf45708085351907aac952a73a7c/corehq/util/workbook_reading/adapters/xlsx.py#L81-L84
via 
https://github.com/dimagi/commcare-hq/blob/c6155325492cdf45708085351907aac952a73a7c/corehq/util/workbook_reading/adapters/generic.py#L9-L19

**Review by nickpell — APPROVED** [review]

**Pull request merged**